src/main/java/Duke.java

```diff
@@ -1,10 +1,108 @@
+import java.util.*;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
+    public static void main(String[] args) throws DukeException {
+        /*String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+         */
+        Scanner sc = new Scanner(System.in);
+        String Hello = "Hello! I'm Duke\nWhat can i do for you?";
+        System.out.println(Hello);
+
+        int flag = 0;
+        ArrayList<task> mylist = new ArrayList<>();
+
+        while(flag == 0) {
+            String input = sc.nextLine();
+            String[] inarr = input.split(" ");
+            String Line = "_________________________________________________________________";
+
+            System.out.println(Line);
+            try {
+                if (input.equals("bye")) {
+                    System.out.println("Bye. Hope to see you again soon!");
+                    flag = 1;
+                } else if (input.equals("list")) {
+                    printlist(mylist);
+                } else if (inarr[0].equals("done")) {
+                    try {
+                        int tocheck = Integer.parseInt(inarr[1]);
+                        task temp = mylist.get(tocheck - 1);
+                        temp.markDone();
+
+                        System.out.println("Nice! I've marked this task as done:");
+                        System.out.println(mylist.get(tocheck - 1));
+                    } catch (IndexOutOfBoundsException e) {
+                        throw new DukeException("☹ OOPS!!! That task doesn't exist or you failed to include a number.", e);
+                    }
+                } else if (inarr[0].equals("todo")) {
+                    try {
+                        ToDo temp = new ToDo(input.replaceFirst("todo ", ""));
+                        System.out.println("Got it. I've added this task:");
+                        System.out.println(temp);
+                        System.out.println("Now you have " + (mylist.size() + 1) + " tasks in the list.");
+                        mylist.add(temp);
+                    } catch (IndexOutOfBoundsException e) {
+                        throw new DukeException("☹ OOPS!!! You're missing some descriptions for your todo.", e);
+                    }
+                } else if (inarr[0].equals("deadline")) {
+                    try {
+                        String wodl = input.replaceFirst("deadline ", "");
```

Comment:
suggest to cleanse and hande raw input in a new Class so that it is less cluttered in your main class